src/views/ProviderView/index.js

```diff
@@ -565,6 +588,10 @@ module.exports = class ProviderView {
       i18n: this.plugin.uppy.i18n
     })
 
-    return <Browser {...browserProps} />
+    return (
+      <CloseWrapper onUnmount={this.clearSelection}>
+        <Browser {...browserProps} />
+      </CloseWrapper>
+    )
```

Comment:
👍

`onUnmount` is handy, but a Preact-only thing it seems. Very much not a big deal, but I was trying to keep an open door for React, hence not using `render(state, props)` and other niceties :) But we can always refactor of course, no need to stick to theoretical benefits :)

Reply:
Naw, `onUnmount` is a plain prop, called inside CloseWrapper's `componentWillUnmount`: https://github.com/transloadit/uppy/pull/826/files#diff-1018535a09fc115f567d294f012f2d2aR19

Reply:
:-D ha-ha, sorry! Googled that prop and found https://github.com/developit/preact/issues/283, it got me confused, I immediately thought they added it to Preact, lol. Sorry.